detectable/src/main/java/com/synopsys/integration/detectable/detectables/poetry/parser/PoetryLockParser.java

```diff
@@ -102,6 +102,14 @@ private Set<String> populatePackageMapAndGetRootPackages(TomlArray lockPackages)
         return rootPackages;
     }
 
+    private void putPackage(String name, Dependency dependency) {
+        packageMap.put(name.toLowerCase(), dependency);
+    }
+
+    private Dependency getPackage(String name) {
+        return packageMap.get(name.toLowerCase());
```

Comment:
Is this redundant? If not in what cases would you expect packageMap to contain a package that has not already been lower cased? Im not familiar with this parser, thanks for clarifying!

Reply:
It's correct that all keys in this `packageMap` are lower-cased.
The goal here is to query the correct value when the `name` variable hasn't been lower-cased.

The overall idea here is:
- packages in the toml files can be specified in any case, potentially inconsistently throughout the file
- we always lower-case the key when storing it in the map for consistency
- we always lower-case the lookup key before attempting to extract a value from the map, again for that consistency